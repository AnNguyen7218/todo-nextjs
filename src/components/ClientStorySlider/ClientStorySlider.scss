@import 'config.gutters';
@import 'config.media';

.container {
  padding-top: gutter(48);
  padding-bottom: gutter(72);

  @include break(xl) {
    padding-top: gutter(24);
    padding-bottom: gutter(96);
  }
}

.title-container {
  margin-bottom: gutter(24);

  @include break(md) {
    margin-bottom: gutter(24);
  }

  @include break(xl) {
    margin-bottom: gutter(48);
  }
}

.swiper-slider {
  height: auto;
}
