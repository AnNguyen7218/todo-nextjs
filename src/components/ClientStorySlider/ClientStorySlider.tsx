import React from 'react';
import { CLIENT_STORIES } from 'constants/sellerLandingPage';
import { Container, Slider, ClientStory } from 'components';
import css from './ClientStorySlider.scss';

const ClientStorySlider = () => {
  return (
    <div className={css['container']}>
      <Container className={css['title-container']}>
        <h2>Our client stories</h2>
      </Container>
      <Container>
        <Slider
          swiperParams={{
            autoplay: {
              delay: 6000,
              disableOnInteraction: false,
            },
            spaceBetween: 24,
            slidesPerView: 1.3,
            breakpoints: {
              720: {
                slidesPerView: 1.5,
              },
            },
          }}
        >
          {CLIENT_STORIES.map((client, index) => {
            const theme = index % 2 === 0 ? 'light' : 'dark';
            return (
              <div key={index} className={css['swiper-slider']}>
                <ClientStory client={client} theme={theme} />
              </div>
            );
          })}
        </Slider>
      </Container>
    </div>
  );
};

export default ClientStorySlider;
