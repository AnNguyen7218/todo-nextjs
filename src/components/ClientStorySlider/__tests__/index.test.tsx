import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ClientStorySlider from '../ClientStorySlider';

afterEach(cleanup);

describe('Slider Component', () => {
  test('Slider should match snapshot', () => {
    const { container } = render(<ClientStorySlider />);
    expect(container).toMatchSnapshot();
  });
});
