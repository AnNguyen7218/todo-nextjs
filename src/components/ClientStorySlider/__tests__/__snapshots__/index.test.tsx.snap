// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component Slider should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container title-container"
    >
      <h2>
        Our client stories
      </h2>
    </div>
    <div
      class="container"
    >
      <div
        class="swiper-container swiper-container-initialized swiper-container-horizontal"
      >
        <div
          class="swiper-wrapper"
          style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide swiper-slider swiper-slide-duplicate"
            data-swiper-slide-index="0"
          >
            <div
              class="container container--light"
            >
              <img
                alt="client-avatar"
                class="avatar"
                src="client1.jpg"
              />
              <p
                class="client-quote"
              >
                "
                Thanks to Intrepid, I now spend more time improving my services instead of searching for customers and chasing leads. The best part is that I'm much less stressed out and have more time for my kids
                "
              </p>
              <p
                class="client"
              >
                Olena Zanichkovska
                ,
                <br />
                <span
                  class="client-title"
                >
                  Founding Partner
                   
                  @The Gradient
                </span>
              </p>
            </div>
          </div>
          <div
            class="swiper-slide swiper-slider swiper-slide-duplicate"
            data-swiper-slide-index="1"
          >
            <div
              class="container container--dark"
            >
              <img
                alt="client-avatar"
                class="avatar"
                src="client2.jpg"
              />
              <p
                class="client-quote"
              >
                "
                We have developed a great relationship with the Intrepid team. Their understanding of the challenges faced by service providers is extremely evident in the platform they have created.
                "
              </p>
              <p
                class="client"
              >
                John Davies
                ,
                <br />
                <span
                  class="client-title"
                >
                  CEO,
                   
                  Symbol
                </span>
              </p>
            </div>
          </div>
          <div
            class="swiper-slide swiper-slider"
            data-swiper-slide-index="0"
          >
            <div
              class="container container--light"
            >
              <img
                alt="client-avatar"
                class="avatar"
                src="client1.jpg"
              />
              <p
                class="client-quote"
              >
                "
                Thanks to Intrepid, I now spend more time improving my services instead of searching for customers and chasing leads. The best part is that I'm much less stressed out and have more time for my kids
                "
              </p>
              <p
                class="client"
              >
                Olena Zanichkovska
                ,
                <br />
                <span
                  class="client-title"
                >
                  Founding Partner
                   
                  @The Gradient
                </span>
              </p>
            </div>
          </div>
          <div
            class="swiper-slide swiper-slider"
            data-swiper-slide-index="1"
          >
            <div
              class="container container--dark"
            >
              <img
                alt="client-avatar"
                class="avatar"
                src="client2.jpg"
              />
              <p
                class="client-quote"
              >
                "
                We have developed a great relationship with the Intrepid team. Their understanding of the challenges faced by service providers is extremely evident in the platform they have created.
                "
              </p>
              <p
                class="client"
              >
                John Davies
                ,
                <br />
                <span
                  class="client-title"
                >
                  CEO,
                   
                  Symbol
                </span>
              </p>
            </div>
          </div>
          <div
            class="swiper-slide swiper-slider swiper-slide-duplicate"
            data-swiper-slide-index="0"
          >
            <div
              class="container container--light"
            >
              <img
                alt="client-avatar"
                class="avatar"
                src="client1.jpg"
              />
              <p
                class="client-quote"
              >
                "
                Thanks to Intrepid, I now spend more time improving my services instead of searching for customers and chasing leads. The best part is that I'm much less stressed out and have more time for my kids
                "
              </p>
              <p
                class="client"
              >
                Olena Zanichkovska
                ,
                <br />
                <span
                  class="client-title"
                >
                  Founding Partner
                   
                  @The Gradient
                </span>
              </p>
            </div>
          </div>
          <div
            class="swiper-slide swiper-slider swiper-slide-duplicate"
            data-swiper-slide-index="1"
          >
            <div
              class="container container--dark"
            >
              <img
                alt="client-avatar"
                class="avatar"
                src="client2.jpg"
              />
              <p
                class="client-quote"
              >
                "
                We have developed a great relationship with the Intrepid team. Their understanding of the challenges faced by service providers is extremely evident in the platform they have created.
                "
              </p>
              <p
                class="client"
              >
                John Davies
                ,
                <br />
                <span
                  class="client-title"
                >
                  CEO,
                   
                  Symbol
                </span>
              </p>
            </div>
          </div>
        </div>
        <span
          aria-atomic="true"
          aria-live="assertive"
          class="swiper-notification"
        />
      </div>
    </div>
  </div>
</div>
`;
